corehq/apps/users/views/utils.py

```diff
@@ -109,3 +110,23 @@ def log_commcare_user_locations_changes(request, user, old_location_id, old_assi
             fields_changed=fields_changed,
             change_messages=change_messages
         )
+
+
+def get_locations_with_cases(domain, location_ids):
+    """
+    Takes a list of location IDs and returns how many cases are assigned to each location
+    :param domain: The domain to search in
+    :param location_ids: A list of location IDs to get case counts on
+    :returns A dict with location names as the key, and the number of cases assigned to it as the value
+    """
+    locations_with_cases = dict()
+    for location_id in location_ids:
+        query = (CaseES()
+                .domain(domain)
+                .owner(location_id))
+        results = query.run().hits
```

Comment:
Instead of running the query and counting the hits, I think you can use the [query's `count()` method](https://github.com/dimagi/commcare-hq/blob/81559b0c8f9dfd921651f05be66f4ce419179fae/corehq/apps/es/es_query.py#L465).
```suggestion
        count = (CaseES()
                .domain(domain)
                .owner(location_id)
                .count())
```

Reply:
can you not do all locations in one go? and use an aggregation to get the count for each?
```
res = CaseES()....aggregation(TermsAggregation('by_location', 'owner_id').size().run()
counts = res.aggregations.by_location.counts_by_bucket()
```

Reply:
That would work better, as then it would only be one query to ES. I have comitted a change as per your suggestion. :+1: